### PR TITLE
Deprecated OpenSSL API fixes.

### PR DESCRIPTION
**Pull request by neheb**
Deprecated API fixes. Again...

### CODE REVIEW

**Comment by chipitsine** [comment]
I will close my PR.

as you are touching this, can you please modify travis-ci script as well (and add "no-deprecated" where appropriate) ?

File: .ci/build-openssl.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ build_openssl () {
     if [[ "$(cat ${OPENSSL_INSTALL_DIR}/.openssl-version)" != "${OPENSSL_VERSION}" ]]; then
         tar zxf "download-cache/openssl-${OPENSSL_VERSION}.tar.gz"
         cd "openssl-${OPENSSL_VERSION}/"
-        ./config shared --prefix="${OPENSSL_INSTALL_DIR}" --openssldir="${OPENSSL_INSTALL_DIR}" -DPURIFY
+        ./config shared no-deprecated --prefix="${OPENSSL_INSTALL_DIR}" --openssldir="${OPENSSL_INSTALL_DIR}" -DPURIFY
```
Comment (chipitsine) [inline .ci/build-openssl.sh]: this will build for every openssl version. even 1.0.2

Comment (chipitsine) [reply .ci/build-openssl.sh]: I did find some issues with openssl-1.0.2 built with no-deprecated

Comment (neheb) [reply .ci/build-openssl.sh]: Compiles now.

LibreSSL doesn't seem to have a good way to disable deprecated APIs. I think build-libressl.sh would need to pass -DOPENSSL_NO_DEPRECATED to CFLAGS. Not sure how to do that.

Comment (neheb) [reply .ci/build-openssl.sh]: I manually checked deprecated LibreSSL APIs. It seems SoftEtherVPN is not using them.

**Pull request merged**